adapters/eplanning/eplanning.go

```diff
@@ -117,7 +118,19 @@ func (adapter *EPlanningAdapter) MakeRequests(request *openrtb.BidRequest, reqIn
 		pageDomain = defaultPageURL
 	}
 
-	uri := adapter.URI + fmt.Sprintf("/%s/%s/%s/%s?r=pbs&ncb=1&ur=%s&e=%s", clientID, dfpClientID, pageDomain, sec, url.QueryEscape(pageURL), strings.Join(spacesStrings, "+"))
+	var requestTarget string
+	if request.App != nil && request.App.Bundle != "" {
+		requestTarget = request.App.Bundle
+	} else {
+		requestTarget = pageDomain
+	}
+
+	var uri string
+	if request.App != nil {
+		uri = adapter.URI + fmt.Sprintf("/%s/%s/%s/%s?r=pbs&ncb=1&e=%s", clientID, dfpClientID, requestTarget, sec, strings.Join(spacesStrings, "+"))
+	} else {
+		uri = adapter.URI + fmt.Sprintf("/%s/%s/%s/%s?r=pbs&ncb=1&ur=%s&e=%s", clientID, dfpClientID, requestTarget, sec, url.QueryEscape(pageURL), strings.Join(spacesStrings, "+"))
+	}
```

Comment:
For a more secure URL build, can we use golang's `"net/url"` library to add the path and the query to the URL?
```
uriObj, err := url.Parse(adapter.URI)
uriObj.Path = fmt.Sprintf("/%s/%s/%s/%s", clientID, dfpClientID, requestTarget, sec)
query := url.Values{}
query.Set("r", "pbs")
query.Set("ncb", 1)
query.Set("e", strings.Join(spacesStrings, "+"))
uriObj.RawQuery = query.Encode()

uri = uriObj.String()
```